### PR TITLE
added conda-bonjour

### PR DESCRIPTION
**Pull request by BolunDai0216**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by BolunDai0216** [comment]
@conda-forge/help-python The recipe has passed all checks and ready for review.

**Review by synapticarbors — COMMENTED** [review]
I'm not sure if this package is really suitable for conda-forge as it is just a "hello world" with dependencies that don't actually match what is in the code. I can't find a particular policy that would exclude this, but I'm trying to understand how this would be used. @conda-forge/staged-recipes do we have a policy on this type of package? It may be ok, but I'm not sure

**Comment by BolunDai0216** [comment]
> I'm not sure if this package is really suitable for conda-forge as it is just a "hello world" with dependencies that don't actually match what is in the code. I can't find a particular policy that would exclude this, but I'm trying to understand how this would be used. @conda-forge/staged-recipes do we have a policy on this type of package? It may be ok, but I'm not sure

@synapticarbors Hi,

Thanks for looking into this. This is just the first step. I plan to expand this package to become a tutorial on how to package conda-forge packages, including C++ source code. I already created a version of this for PyPI. See https://github.com/BolunDai0216/pypi-bonjour.

File: recipes/conda-bonjour/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{% set name = "conda-bonjour" %}
+{% set version = "0.0.1" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/BolunDai0216/conda-bonjour/archive/v{{ version }}.tar.gz
```
[the remark this answers is not in the document]
Comment (BolunDai0216) [reply recipes/conda-bonjour/meta.yaml:9]: Hi, 

@ocefpaf Thanks for looking into this.

I'm afraid I have to disagree with this statement; tutorials are meaningful, and right now, they are lacking. At least I did not find a package that could show me how to publish a Python + C++ package to `conda-forge` that I could quickly parse. There are many packages on `conda-forge` that do have Python bindings of C++ source code, but almost all of them have many other things that are not related to `conda` going on, which makes it difficult for someone to quickly figure out whats important for publishing to `conda-forge` and what is not.

[the remark this answers is not in the document]
Comment (BolunDai0216) [reply recipes/conda-bonjour/meta.yaml:9]: Nevermind, I'll do this in my own channel, which I assume once that works it should also work on conda-forge.

**Pull request closed without merging**